apps/events/api/views.py

```diff
@@ -200,7 +200,11 @@ def _get_allowed_attendees(user):
         return attendees.distinct()
 
     def get_queryset(self):
-        return self._get_allowed_attendees(self.request.user)
+        return (
+            Attendee.objects.none()
+            if self.request.user.is_anonymous
+            else self._get_allowed_attendees(self.request.user)
```

Comment:
I think the right move to make here is to add `IsAuthenticated` permission to the viewset?

If not then maybe move the check as an early return in the `_get_allowed_attendees` method?

Reply:
Since this function has it's own permission-classes for every method, it's much less work to do it here. As you suggested, I moved it to `_get_allowed_attendees`